x-pack/plugins/observability/public/pages/rule_details/utils.ts

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { RuleExecutionStatusValues } from '@kbn/alerting-plugin/common';
+
+export const getColorStatusBased = (ruleStatus: string) => {
```

Comment:
There is already a similar function in triggers_actions_ui https://github.com/elastic/kibana/blob/main/x-pack/plugins/triggers_actions_ui/public/application/sections/rules_list/components/rule_execution_status_filter.tsx#L92, so we'd better reuse stuff from there, otherwise we would deviate from Stack Management Rule Detail page if they changed something.

I am fine to address this issue in a separate PR, but we should definitely keep share-ability in mind. I did something similar in Rules page and defined some functions under Observability (`getHealthColor` is one of them), but I plan to replace these functions with reusable ones from triggers_actions_ui plugin.

Reply:
@mgiota, I Agree. For now, I will use the one you added for the Rules page, which is identical to the one you mentioned as it's easier to import. 
In a separate PR, this function should be placed in a generic place in triggers_actions_ui (out of rule_execution_status_filter) then we can import it.